tests/strategies/forward/test_strategies.py

```diff
@@ -0,0 +1,196 @@
+    """Fixture for the ForwardStrategy classes."""
+    ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS = True
+    orig_start_date = GlobalConfig.start_date
+    strategy = request.param[0](capacity_kW=100, order_updater_parameters=request.param[1])
+    strategy_area = Area("asset", strategy=strategy)
+    area = Area("grid", children=[strategy_area])
+    area.config.start_date = CURRENT_MARKET_SLOT
+    area.config.end_date = area.config.start_date.add(years=20)
+    yield strategy, area
+    ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS = False
+    GlobalConfig.start_date = orig_start_date
+
+
+class TestForwardStrategies:
+    # pylint: disable=protected-access
+    @staticmethod
+    def _assert_posted_orders_on_markets(strategy, market_type, order_count, energy_rate, energy):
+        intraday = strategy.area.forward_markets[market_type]
+        if isinstance(strategy, ForwardLoadStrategy):
+            orders = intraday.get_bids().values()
+        else:
+            orders = intraday.get_offers().values()
+
+        assert len(orders) == order_count
+        for order in orders:
+            assert order.energy_rate == energy_rate
+            assert order.energy == energy
+            assert order.buyer == order.buyer_origin == strategy.owner.name
+            assert order.buyer_id == order.buyer_origin_id == strategy.owner.uuid
+
+    @staticmethod
+    @pytest.mark.parametrize("market_type, expected_order_updater_count, next_slot_timestamp, ", [
+        (AvailableMarketTypes.INTRADAY, 24 * 4 - 1,
+         CURRENT_MARKET_SLOT.add(minutes=15)),
+        (AvailableMarketTypes.DAY_FORWARD, 24 * 7 - 1,
+         CURRENT_MARKET_SLOT.add(days=1)),
+        (AvailableMarketTypes.WEEK_FORWARD, 51,
+         CURRENT_MARKET_SLOT.start_of("week").add(weeks=1)),
+        (AvailableMarketTypes.MONTH_FORWARD, 23,
+         CURRENT_MARKET_SLOT.start_of("month").add(months=1)),
+        (AvailableMarketTypes.YEAR_FORWARD, 5,
+         CURRENT_MARKET_SLOT.start_of("year").add(years=1)),
+    ])
+    def test_order_updaters_follow_market_slots(
+            forward_strategy_fixture, market_type,
+            expected_order_updater_count, next_slot_timestamp):
+        strategy = forward_strategy_fixture[0]
+        area = forward_strategy_fixture[1]
+        area.activate()
+        strategy.event_market_cycle()
+        market_object = area.forward_markets[market_type]
+        assert (
+            list(strategy._order_updaters[market_object].keys()) ==
+            market_object.market_time_slots)
+        assert (
+            len(strategy._order_updaters[market_object].keys()) ==
+            expected_order_updater_count)
+
+        with patch("gsy_e.models.area.Area.now", new_callable=PropertyMock) as now_mock:
+            now_mock.return_value = next_slot_timestamp
+            area.cycle_markets()
+            # strategy.event_market_cycle()
```

Comment:
this is not needed, because you are doing it in line 92 already right? Otherwise I would expect this to be needed.

Reply:
Missed that, it is indeed not needed. I initially wanted to cycle only the strategy, but during the implementation it was proven that some parts of the Area class need to be cycled too. Will change this.